modyn/supervisor/internal/pipeline_executor/pipeline_executor.py

```diff
@@ -717,110 +726,89 @@ def _evaluate_and_store_results(
             pipeline_stage_msg(PipelineStage.EVALUATE, MsgType.ID, id_submsg(IdType.TRIGGER, trigger_id))
         )
 
-        # let's add the business logic for the new eval handler setup in a follow-up PR
-        eval_strategy_config = self.state.pipeline_config.evaluation.handlers[0].strategy
-        eval_strategy_module = dynamic_module_import("modyn.supervisor.internal.eval.strategies")
-        eval_strategy: AbstractEvalStrategy = getattr(eval_strategy_module, eval_strategy_config.type)(
-            eval_strategy_config.model_dump(by_alias=True)
-        )
+        eval_requests: list[EvalRequest] = []
 
-        for eval_dataset_config in self.state.pipeline_config.evaluation.datasets:
-            for interval_start, interval_end in eval_strategy.get_eval_intervals(first_timestamp, last_timestamp):
-                results = self._single_evaluation(
-                    s, self.logs, trigger_id, training_id, model_id, eval_dataset_config, interval_start, interval_end
-                )
-                if results:
-                    self._store_evaluation_results(
-                        s,
-                        self.logs,
-                        trigger_id,
-                        model_id,
-                        eval_dataset_config.dataset_id,
-                        results,
-                        interval_start,
-                        interval_end,
-                    )
+        for eval_handler in self.eval_handlers:
+            if eval_handler.config.execution_time != "after_training":
+                continue
+
+            handler_eval_requests = eval_handler.get_eval_requests_after_training(
+                trigger_id=trigger_id,
+                training_id=training_id,
+                model_id=model_id,
+                training_interval=(first_timestamp, last_timestamp),
+            )
+            eval_requests += handler_eval_requests
+
+        # will be replaced with threadpool in a follow-up PR
```

Comment:
I am a bit torn whether we need this on main. On the one hand, for now this greatly speeds up the results because we paralleliez eval, on the other hand, when we parallelize eval in the future we want to do it at the evaluator itself, and then we do not need the thread pool anymore but instead can just submit requests in a quick loop

Reply:
In the next PR i will move the evaluation logic into an `evaluation_executor.py` so it doesn't mess up the `pipeline_executor.py`. There we can simply remove that again in the future, it will only be a couple of lines in a dedicated function, that is responsible for executing a list of evaluation requests.